spec/models/audio_event_spec.rb

```diff
@@ -508,6 +691,82 @@
     expect(actual_audio_event_ids).to eq(expected_audio_event_ids)
   end
 
+  describe 'verifications query for annotation downloads' do
+    create_entire_hierarchy
+
+    let(:event_one) { create(:audio_event, creator: writer_user, audio_recording:) }
+
+    let(:tag_one) { create(:tag_taxonomic_true_common, creator: writer_user) }
+    let(:tag_two) { create(:tag_taxonomic_true_common, creator: writer_user) }
+
+    let(:tagging_one) {
+      create(:tagging,
+        audio_event_id: event_one.id,
+        tag_id: tag_one.id,
+        creator: writer_user)
+    }
+    let(:tagging_two) {
+      create(:tagging,
+        audio_event_id: event_one.id,
+        tag_id: tag_two.id,
+        creator: writer_user)
+    }
+
+    let(:choices_one) { { 'correct' => 4, 'incorrect' => 3, 'skip' => 2, 'unsure' => 1 } }
+    let(:choices_two) { { 'correct' => 1, 'incorrect' => 4, 'skip' => 3, 'unsure' => 2 } }
+
+    before do
+      event_verifications_for_tagging(tagging_one, choices_one)
+      event_verifications_for_tagging(tagging_two, choices_two)
+    end
+
+    it 'aggregates verifications correctly' do
+      verification_cte_table, query_cte = AudioEvent.verification_summary_cte
+
+      query = AudioEvent.arel_table.join(verification_cte_table)
+        .on(AudioEvent.arel_table[:id].eq(verification_cte_table[:audio_event_id]))
+        .project(verification_cte_table[Arel.star])
+        .with(query_cte)
+
+      formatted_annotations = AudioEvent.connection.select_all(query.to_sql)
+
+      expect(formatted_annotations.pluck('verifications')).to eq([
+        "#{tag.id}:#{tag.text}",
+        "#{tag_one.id}:#{tag_one.text}|#{tag_two.id}:#{tag_two.text}"
+      ])
+
+      combined_counts = [choices_one.values.sum, choices_two.values.sum].join('|')
+      combined_correct = combine_for_key('correct', choices_one, choices_two)
+      combined_incorrect = combine_for_key('incorrect', choices_one, choices_two)
+      combined_skip = combine_for_key('skip', choices_one, choices_two)
+      combined_unsure = combine_for_key('unsure', choices_one, choices_two)
+
+      combined_decisions = [
+        majority_decision(choices_one),
+        majority_decision(choices_two)
+      ].join('|')
+
+      combined_consensus = [
+        consensus_for(choices_one),
+        consensus_for(choices_two)
+      ].join('|')
+
+      aggregated_fields = {
+        'verification_counts' => ['1', combined_counts],
+        'verification_correct' => ['1', combined_correct],
+        'verification_incorrect' => ['0', combined_incorrect],
+        'verification_skip' => ['0', combined_skip],
+        'verification_unsure' => ['0', combined_unsure],
+        'verification_decisions' => ['correct',    combined_decisions],
+        'verification_consensus' => ['1.00',       combined_consensus]
+      }
+
+      aggregated_fields.each do |field_name, expected|
+        expect(formatted_annotations.pluck(field_name)).to eq(expected)
+      end
```

Comment:
I would aim to use fewer helper functions here. When you're testing code you don't want too many helpers, otherwise you end up testing the helpers.

Your test data should be stable enough that you can just use string templating.

I was going to try and provide an example of what I mean, but I can't work out the meaning of `aggregated_fields`